iphone/Maps/Bookmarks/BookmarksList/BookmarksListViewController.swift

```diff
@@ -122,12 +127,30 @@ extension BookmarksListViewController: UITableViewDelegate {
   func tableView(_ tableView: UITableView, didEndEditingRowAt indexPath: IndexPath?) {
     isEditing = false
   }
+  
+  func tableView(_ tableView: UITableView,
+                 leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+    let moveAction = UIContextualAction(style: .normal, title: L("Move")) { [weak self] (_, _, completion) in
+      guard let section = self?.sections?[indexPath.section] else { fatalError() }
+      self?.presenter.moveItem(in: section, at: indexPath.row)
+      completion(true)
+    }
+    return UISwipeActionsConfiguration(actions: [moveAction])
+  }
 
   func tableView(_ tableView: UITableView,
-                 commit editingStyle: UITableViewCell.EditingStyle,
-                 forRowAt indexPath: IndexPath) {
-    guard let section = sections?[indexPath.section] else { fatalError() }
-    presenter.deleteBookmark(in: section, at: indexPath.row)
+                 trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+    let deleteAction = UIContextualAction(style: .destructive, title: L("Delete")) { [weak self] (_, _, completion) in
+      guard let section = self?.sections?[indexPath.section] else { fatalError() }
+      self?.presenter.deleteItem(in: section, at: indexPath.row)
+      completion(true)
+    }
+    let editAction = UIContextualAction(style: .normal, title: L("Edit")) { [weak self] (_, _, completion) in
+      guard let section = self?.sections?[indexPath.section] else { fatalError() }
+      self?.presenter.editItem(in: section, at: indexPath.row)
+      completion(true)
+    }
+    return UISwipeActionsConfiguration(actions: [deleteAction, editAction])
```

Comment:
Move, Delete, Edit strings are localized by default or this actions are not drawable?

Reply:
Good catch, thank you.

- "delete" already exists in `strings.txt`, only the key name starts from `d`, not `D`, I've changed key name in code.
- "edit" was only for android, I've added `ios` tag.
- there were no "move", I've added it too.

I have few questions about localization process:
1. How can I generate localized.strings from updated string.txt to test it locally? 
2. If I'll generate localized.strings from strings.txt should I commit it?
3. What should I do with new key translations for the languages which I don't know? Should I add all languages to strings.txt anyway and set english translations for all other languages, or should I keep the translations empty, or, maybe I shouldn't add other languages to new key at all and english will be used by default?

Thanks!

Reply:
Seems like you already did well with strings!